admiral/pkg/apis/admiral/routes/handlers.go

```diff
@@ -25,4 +38,94 @@ func (opts *RouteOpts) ReturnSuccessGET(w http.ResponseWriter, r *http.Request)
 
 func (opts *RouteOpts) GetClusters(w http.ResponseWriter, r *http.Request) {
 
+	var clusterList []string
+
+	// loop through secret controller's c.cs.remoteClusters to access all clusters admiral is watching
+	for clusterID := range opts.RemoteRegistry.SecretController.Cs.RemoteClusters {
+		clusterList = append(clusterList, clusterID)
+	}
+
+	out, err := json.Marshal(clusterList)
+	if err != nil {
+		log.Printf("Failed to marshall response")
+		http.Error(w, "Failed to marshall response", http.StatusInternalServerError)
+	} else {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(200)
+		w.Write(out)
+	}
+}
+
+func (opts *RouteOpts) GetServiceEntriesByCluster(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	params := mux.Vars(r)
+	clusterName := strings.Trim(params["clustername"], " ")
+
+	var response []v1alpha3.ServiceEntry
+
+	if clusterName != "" {
+
+		serviceEntriesByCluster, err := clusters.GetServiceEntriesByCluster(clusterName, opts.RemoteRegistry)
+
+		if err != nil {
+			log.Printf(err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		} else {
+			if len(serviceEntriesByCluster) == 0 {
+				log.Printf(fmt.Sprintf("No service entries configured for cluster - %s", clusterName))
+				http.Error(w, fmt.Sprintf("No service entries configured for cluster - %s", clusterName), http.StatusNotFound)
```

Comment:
I think the behavior should be:
The cluster doesn't exist (is not being monitored by admiral) -> 404
The cluster has 0 service entries -> empty response object with 200

Reply:
Makes sense, updated the response handling